lib/test.js

```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
     this.pendingCount = 0;
     this._skip = args.opts.skip || false;
     this._timeout = args.opts.timeout;
-    this._objectPrintDepth = args.opts.objectPrintDepth || 5;
     this._plan = undefined;
     this._cb = args.cb;
     this._progeny = [];
     this._ok = true;
+    var depthEnvVar = process.env.NODE_TAPE_OBJECT_PRINT_DEPTH;
+    if (args.opts.objectPrintDepth) {
+        this._objectPrintDepth = args.opts.objectPrintDepth;
+    } else if (depthEnvVar) {
+        if (depthEnvVar.toLowerCase() === 'infinity') {
```

Comment:
To clarify: we should have `var toLowerCase = String.prototype.toLowerCase;` near the top of the file, and then here, do `toLowerCase.call(depthEnvVar)` instead of assuming that nobody's done `delete String.prototype.toLowerCase` inside their tests.